src/Avalonia.Controls/Presenters/ContentPresenter.cs

```diff
@@ -327,7 +327,11 @@ protected virtual IControl CreateChild()
             var oldChild = Child;
             var newChild = content as IControl;
 
-            if (content != null && newChild == null)
+            // We want to allow creating Child from the Template, if Content is null.
+            // But it's important to not inherit data template in this case nor use DataTemplates,
```

Comment:
What does "inherit data template" mean in this case?

Reply:
A bit out-of-dated comment. Previous code allowed to use DataTemplates defined on this control, but ignore DataTemplates from parents (inherited).

Reply:
But here we ignore DataTemplates completely now (otherwise there are tons of broken places). So this comment should be updated.
```suggestion
            // But it's important to use DataTemplates, otherwise we will break content presenters in many places,
```

Reply:
Shouldn't "use DataTemplates" in your edit be "not use DataTemplates"?